src/_pytest/pytester.py

```diff
@@ -760,6 +761,9 @@ def _makefile(
     ) -> Path:
         items = list(files.items())
 
+        if ext is None:
```

Comment:
Let's leave a note here:

```suggestion
        # https://github.com/python/cpython/issues/119113  
        if ext is None:
```

Reply:
I don't think it's necessary, since I think we better not rely on it and just keep the check.

Reply:
I don't see what harm a small comment like that could cause either, but OK.

Reply:
Also in that case, we probably should change the signature as well, so we have the typing match the runtime behavior.

Reply:
The type is `ext: str` which is correct. The callers have some `*args, **kwargs` forwarding which is somewhat annoying to type.

Reply:
Well OK, this is no big deal anyway.